Document provider: allow rename

tobiasKaminsky opened the pull request:
- use >=5.0 device
- open files/documents app
- long press on file/folder 
- rename

AndyScherzinger commented:
@tobiasKaminsky could you rebase & resolve conflicts please ❤️

AndyScherzinger commented:
@tobiasKaminsky at least on a not downloaded file I get:
```
2019-01-16 13:58:27.252 17321-17536/com.nextcloud.client E/DatabaseUtils: Writing exception to parcel
    android.os.ParcelableException: java.io.FileNotFoundException: Failed to rename document with documentId 3
        at android.provider.DocumentsProvider.call(DocumentsProvider.java:953)
        at android.content.ContentProvider$Transport.call(ContentProvider.java:403)
        at android.content.ContentProviderNative.onTransact(ContentProviderNative.java:272)
        at android.os.Binder.execTransact(Binder.java:731)
     Caused by: java.io.FileNotFoundException: Failed to rename document with documentId 3
        at com.owncloud.android.providers.DocumentsStorageProvider.renameDocument(DocumentsStorageProvider.java:320)
        at android.provider.DocumentsProvider.callUnchecked(DocumentsProvider.java:1026)
        at android.provider.DocumentsProvider.call(DocumentsProvider.java:951)
        at android.content.ContentProvider$Transport.call(ContentProvider.java:403) 
        at android.content.ContentProviderNative.onTransact(ContentProviderNative.java:272) 
        at android.os.Binder.execTransact(Binder.java:731) 
2019-01-16 13:58:39.095 2147-2343/? E/storaged: getDiskStats failed with result NOT_SUPPORTED and size 0

```

AndyScherzinger commented:
@tobiasKaminsky downloaded files do work (e.g. editing a file first)

tobiasKaminsky commented:
Hm.
I tried it and it worked in both cases, even for folders.
Also with whitespaces, special chars in folder/file name.

I added exception to the error, maybe we can now see what the problem is.
(I also rebased with latest master, so that refresh is included)
